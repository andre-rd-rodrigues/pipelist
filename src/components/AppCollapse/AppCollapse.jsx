import React, { useState } from "react";
import styles from "./appcollapse.module.scss";

function AppCollapse({ children }) {
  const initialLabel = "See more";

  const [label, setLabel] = useState(initialLabel);

  return (
    <div className={styles.collapse}>
      <button
        type="button"
        data-bs-toggle="collapse"
        data-bs-target="#appCollapse"
        aria-expanded="false"
        aria-controls="appCollapse"
        className={styles.button}
        onClick={() =>
          setLabel(label === initialLabel ? "Close" : initialLabel)
        }
      >
        {label}
      </button>
      <div className="collapse" id="appCollapse">
        {children}
      </div>
    </div>
  );
}

export default AppCollapse;
