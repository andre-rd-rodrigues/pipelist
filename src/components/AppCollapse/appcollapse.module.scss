@import "styles/mixins";

.collapse {
  margin: 1.5rem 0;
}
.button {
  display: block;
  margin: auto;
  border: none;
  background: none;
  color: $grey;
  font-weight: bold;
  &:hover {
    color: grey;
  }
}
