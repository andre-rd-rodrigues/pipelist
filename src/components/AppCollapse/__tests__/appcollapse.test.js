import { fireEvent, render, screen } from "@testing-library/react";
import reactTestRenderer from "react-test-renderer";
import AppCollapse from "../AppCollapse";

describe("App collapse", () => {
  it("renders correctly", () => {
    const wrapper = reactTestRenderer.create(<AppCollapse />);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders correct initial label", () => {
    render(<AppCollapse />);

    expect(screen.getByText(/See more/i)).toBeInTheDocument();
  });

  it("renders correct label after clicking", () => {
    render(<AppCollapse />);

    const buttonElement = screen.getByRole("button");

    fireEvent.click(buttonElement);
    expect(screen.getByText(/Close/i)).toBeInTheDocument();
  });
});
