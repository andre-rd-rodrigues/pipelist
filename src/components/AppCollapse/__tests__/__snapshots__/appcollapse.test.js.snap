// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App collapse renders correctly 1`] = `
<div
  className="collapse"
>
  <button
    aria-controls="appCollapse"
    aria-expanded="false"
    className="button"
    data-bs-target="#appCollapse"
    data-bs-toggle="collapse"
    onClick={[Function]}
    type="button"
  >
    See more
  </button>
  <div
    className="collapse"
    id="appCollapse"
  />
</div>
`;
